cmd/multiclusterapp.go

```diff
@@ -0,0 +1,854 @@
+	var version string
+	if cachedVersion, ok := templateVersionCache[ID]; ok {
+		version = cachedVersion
+	} else {
+		templateVersion, err := client.TemplateVersion.ByID(ID)
+		if err != nil {
+			return "", err
+		}
+		templateVersionCache[templateVersion.ID] = templateVersion.Version
+		version = templateVersion.Version
+	}
+	return version, nil
+}
+
+func getClusterProjectMap(ctx *cli.Context, client *managementClient.Client) (map[string]managementClient.Cluster, map[string]managementClient.Project, error) {
+	clusters := map[string]managementClient.Cluster{}
+	clusterCollectionData, err := listAllClusters(ctx, client)
+	if err != nil {
+		return nil, nil, err
+	}
+	for _, c := range clusterCollectionData {
+		clusters[c.ID] = c
+	}
+	projects := map[string]managementClient.Project{}
+	projectCollectionData, err := listAllProjects(ctx, client)
+	if err != nil {
+		return nil, nil, err
+	}
+	for _, p := range projectCollectionData {
+		projects[p.ID] = p
+	}
+	return clusters, projects, nil
+}
+
+func listAllClusters(ctx *cli.Context, client *managementClient.Client) ([]managementClient.Cluster, error) {
+	clusterCollection, err := client.Cluster.List(defaultListOpts(ctx))
+	if err != nil {
+		return nil, err
+	}
+	clusterCollectionData := clusterCollection.Data
+	for {
+		clusterCollection, _ = clusterCollection.Next()
+		if clusterCollection == nil {
+			break
+		}
+		clusterCollectionData = append(clusterCollectionData, clusterCollection.Data...)
+		if !clusterCollection.Pagination.Partial {
+			break
+		}
+	}
+	return clusterCollectionData, nil
+}
+
+func listAllProjects(ctx *cli.Context, client *managementClient.Client) ([]managementClient.Project, error) {
+	projectCollection, err := client.Project.List(defaultListOpts(ctx))
+	if err != nil {
+		return nil, err
+	}
+	projectCollectionData := projectCollection.Data
+	for {
+		projectCollection, _ = projectCollection.Next()
+		if projectCollection == nil {
+			break
+		}
+		projectCollectionData = append(projectCollectionData, projectCollection.Data...)
+		if !projectCollection.Pagination.Partial {
+			break
+		}
+	}
+	return projectCollectionData, nil
+}
+
+func getReadableTargetNames(clusterCache map[string]managementClient.Cluster, projectCache map[string]managementClient.Project, targets []managementClient.Target) []string {
+	var targetNames []string
+	var err error
+	for _, target := range targets {
+		projectID := target.ProjectID
+		clusterID, _ := parseScope(projectID)
+		cluster, ok := clusterCache[clusterID]
+		if !ok {
+			logrus.Debugf("Cannot get readable name for target %q: %v\nShow the target in ID instead.", target.ProjectID, err)
+			targetNames = append(targetNames, target.ProjectID)
+			continue
+		}
+		project, ok := projectCache[projectID]
+		if !ok {
+			logrus.Debugf("Cannot get readable name for target %q: %v\nShow the target in ID instead.", target.ProjectID, err)
```

Comment:
Drop `err` from this and the newline, try to keep logs to a single line. `"Cannot get readable name for target %q, showing ID"`